### PR TITLE
Add Jackal user manual, integration, and tutorials

### PR DESCRIPTION
**Pull request by jhiggins-cpr**
Found and fixed minor issues in several other components along the way.

### CODE REVIEW

File: docs/robots/dingo/maintenance_dingo.mdx [hunk 1]
```diff
@@ -4,6 +4,7 @@ sidebar_position: 5
 ---
 
 import ComponentIntroductionDingo from "/components/introduction_dingo.mdx";
+import ComponentLeadAcidBatteryCare from "/components/maintenance/lead_acid_battery_care.mdx";
```
Comment (rhysfaultless-cpr) [inline docs/robots/dingo/maintenance_dingo.mdx:7]: This Component is not being called anywhere. Should this line still be included in the Pull Request?

File: docs/robots/jackal/integration_jackal.mdx [hunk 2]
```diff
@@ -1,12 +1,98 @@
 ---
-title: Integration
+title: Jackal Integration
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/integration_jackal.mdx:2]: This should just be 'Integration' to be aligned with our other robots.

Comment (rhysfaultless-cpr) [reply docs/robots/jackal/integration_jackal.mdx:2]: Or, add a 'sidebar_label' line to the YAML header.

---
title: Jackal Integration
sidebar_label: Integration
sidebar_position: 2
toc_min_heading_level: 2
toc_max_heading_level: 4
---

File: docs/robots/jackal/integration_jackal.mdx [hunk 3]
```diff
@@ -1,12 +1,98 @@
...
+    <figcaption>User power board</figcaption>
+  </figure>
+</center>
+
+#### User Power Connections
+
+Similar to the data connections, power leads may be brought through the hatch, and connected to the User Power Board.
+The following are available as user power:
+
+- unregulated VBAT power (25.6 V nominal, 16 A max), shared across the High Current VBAT User Power (16 A max),
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/integration_jackal.mdx]: Add a capital U for 'Unregulated'

File: docs/robots/jackal/integration_jackal.mdx [hunk 4]
```diff
@@ -1,12 +1,98 @@
...
+  User Power Ports (8 A max per port) and the User Power Screw Terminals (15 A max)
+- regulated 12 V power (7 A max), shared across the User Power ports (7 A max per port) and the User Power
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/integration_jackal.mdx]: Add a capital R for 'Regulated'

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 5]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+#### Pairing the Controller {#pairing-the-controller}
+
+##### PS4 Controller
+
+<ComponentPs4ControllerPairing />
+
+### Setting up Jackal's Network Configuration {#jackal-networking}
+
+Jackal is normally equipped with a combination Wi-Fi + Bluetooth module.
+If this is your first unboxing, ensure that Jackal's wireless antennae are firmly screwed on to the chassis.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Change to 
'...screwed onto the chassis.'
or
'...screwed to the chassis.'

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 6]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+[Gazebo](https://gazebosim.org/home) is the most common simulation tool used in ROS. Jackal's model in Gazebo include reasonable approximations of its dynamics, including wheel slippage, skidding, and inertia.
+To launch simulated Jackal in a simple example world, run the following command:
+
+```
+roslaunch jackal_gazebo jackal_world.launch
+```
+
+You should see the following window appear, or something like it.
+You will see a base Jackal spawned with no additional sensors.
+You can adjust the camera angle by clicking and dragging while holding CTRL, ALT, or the shift key.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Change to 'Shift'

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 7]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+1.  Using the provided PS4 controller.
+    Refer to the [User Manual](/docs/robots/jackal/user_manual_jackal#controller) details on how to use the controller.
+
+2.  Using the RViz instance above.
+    If you select the Interact option in the top toolbar, an interactive marker will appear around the Jackal and can be used to control speed.
+
+3.  The [rqt_robot_steering plugin](http://wiki.ros.org/rqt_robot_steering).
+    Run the `rqt` command, and select **Plugins→Robot Tools→Robot Steering** from the top menu.
+
+Jackal uses [twist_mux](http://wiki.ros.org/twist_mux) to mix separate [geometry_msgs\Twist](http://docs.ros.org/api/geometry_msgs/html/msg/Twist.html) control channels into the `jackal_velocity_controller/cmd_vel` topic.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Change to a forward slash, 'geometry_msgs/Twist'

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 8]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+### Extending Jackal Startup
+
+Now that you've had Jackal for a while, you may be interested in how to extend it, perhaps add some more payloads, or augment the URDF.
+
+#### Startup Launch Context
+
+When ROS packages are grouped together in a directory and then built as one, the result is referred to as a workspace.
+Each workspace generates a `setup.bash` file which the user may source in order to correctly set up important environment variables such as `PATH`, `PYTHONPATH`, and `CMAKE_PREFIX_PATH`.
+
+The standard system-wide setup file is in `opt`:
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: change to '/opt'    ?

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 9]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+
+Each Jackal leaves the factory already configured to pull packages from http://packages.ros.org as well as http://packages.clearpathrobotics.com.
+To update your package and download new package versions, simply run:
+
+```
+sudo apt-get update
+sudo apt-get dist-upgrade
+```
+
+#### MCU Firmware Update
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Should this be in Tutorials or in Maintenance?
MCU firmware seems to blue the line between using Noetic packages on the robot, and maintaining the robots functionality.

As a similar situation; we have Puma firmware updates in the Maintenance section of Husky.

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 10]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+
+You should see about 20 seconds worth of lines output beginning with `Download from image ...`. When this is
+complete perform the following steps to restore normal operation:
+
+1. Move the mode switch back to `NORM` (left).
+2. Press `M_RST`. You must perform this step within a few seconds of changing the mode switch. If you take too long Jackal may power down. This is fine and won't harm the robot, but can be annoying.
+
+If the robot powers-down, simply ensure the mode switch is in the `NORM` position and power the robot back on.
+
+Jackal will now be using the new firmware you installed.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Should we mention moving MCU switches back to their original positions, or rebooting the robot?

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 11]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+
+:::
+
+Referring to the [Simulating Jackal](#simulating-jackal) instructions, bring up Jackal with the front laser enabled for the following demos:
+
+```
+roslaunch jackal_gazebo jackal_world.launch config:=front_laser
+```
+
+If you're working with a real Jackal, it's suggested to connect via SSH and launch the [jackal_navigation](https://github.com/jackal/jackal/tree/noetic-devel/jackal_navigation) launch files from on board the robot.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Change to 'onboard'

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 12]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
+  :::
+
+- **Drive Test**
+
+  The **Drive Test** drives the robot forward 1 metre and checks that the motors, encoders, and encoder-fused odometry are working properly.
+
+  This test:
+
+  - Subscribes to the `/jackal_velocity_controller/odom` ROS topic to receive linear displacement estimations from the encoder-fused odometry.
+    The robot will drive forward until 1 metre is estimated.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx:764]: This file has 'metre' and 'meter'.
Please choose a standard going forward, and consider adding a common spellings Do/Dont section to the README.

| Preferred spelling | Don't |
| :------------------- | :------ |
| Metre                    | Meter |
| RViz                       | Rviz    |

File: docs/robots/jackal/tutorials_jackal.mdx [hunk 13]
```diff
@@ -4,9 +4,1317 @@ sidebar_position: 3
...
 
 :::
+
+:::note
+
+The Jackal ROS Nodes and support software is distributed as source code on Github. Other Clearpath repositories have not been enabled as of April 2020, but will be evaluated based on demand.
+
+:::
+
+#### MCU Firmware Update
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/tutorials_jackal.mdx]: Line 574 of this file has similar MCU Firmware Update instructions. Can these be condensed into a single section?

Comment (rhysfaultless-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: If not; consider adding some reference to Windows 10 for all the subheadings to reduce confusion for people skimming the page: 
'Installing Jackal Software, on Windows 10'
'MCU Firmware Update, on Windows 10'

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 14]
```diff
@@ -1,12 +1,358 @@
...
+
+<ComponentSafetyWarning />
+
+### Shipment Contents
+
+Your Jackal shipment contains the following:
+
+- Clearpath Robotics Jackal
+- 270 Wh lithium battery pack
+- 110V/220V universal charger
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Prefered:
110 VAC - 220 VAC battery charger

or 
~110 V - 220 V battery charger
or
110 V - 220 V battery charger

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 15]
```diff
@@ -1,12 +1,358 @@
...
+
+If you purchased standard payload modules or custom integration services with Jackal, then additional
+equipment will be included per your specific configuration, plus further documentation as required.
+
+### Hardware Overview
+
+#### System Architecture
+
+Jackal is built around a computer running Ubuntu (Intel-based computer or Jetson developer kit), paired
+with a 32-bit MCU. The MCU handles IO, power supply monitoring and motor control, as well as supplying
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Suggest this be changed to:
'with a 32-bit microcontroller MCU.

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 16]
```diff
@@ -1,12 +1,358 @@
...
+in the `jackal_base` node, where it is connected to Jackal's kinematic controller.
+
+An internal magnetometer is available, but not recommended for use due to its location in the vehicle chassis.
+
+#### Exterior Features
+
+The Jackal exterior is shown below and includes:
+
+- 80 mm mounting pattern on the lid
+- 190 mm diamter wheels
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: 'diameter'

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 17]
```diff
@@ -1,12 +1,358 @@
...
+- human machine interface panel (HMI)
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Human

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 18]
```diff
@@ -1,12 +1,358 @@
...
+- lid latches.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Lid

and, remove the period

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 19]
```diff
@@ -1,12 +1,358 @@
...
+documention portal, under the [Computers](/docs/computers), [Sensors](/docs/sensors), [Manipulators](/docs/manipulators),
+and [Accessories](/docs/accessories) sections. In addition, refer to [Jackal Integration](/docs/robots/jackal/integration_jackal) for
+details on how to integrate your own payload or how to integrate a payload kit from Clearpath Robotics.
+Each payload kit that ships from Clearpath Robotics comes with a custom mounting bracket and cabling for
+easy attachment to Jackal.
+
+#### Interior Features
+
+To access Jackal's interior, actuate the latches under the front end of the lid, on the opposite end from
+the HMI. When you lift the lid, you will see Jackal's onboard Li-Ion battery pack, and its two connectors.
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Suggest changing Li-Ion refereces to 'lithium'

File: docs/robots/dingo/maintenance_dingo.mdx (hunk 1 above)
Comment (jhiggins-cpr) [reply docs/robots/dingo/maintenance_dingo.mdx:7]: It's being used on line 17 in this file (at least from what I can see)

File: docs/robots/jackal/integration_jackal.mdx (hunk 2 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/integration_jackal.mdx:2]: Fixed

File: docs/robots/jackal/integration_jackal.mdx (hunk 3 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/integration_jackal.mdx]: Done

File: docs/robots/jackal/integration_jackal.mdx (hunk 4 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/integration_jackal.mdx]: Done

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 5 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Done

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 6 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Done

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 7 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Done

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 8 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Done

File: docs/robots/jackal/user_manual_jackal.mdx [hunk 20]
```diff
@@ -1,12 +1,358 @@
...
+
+### Using ROS
+
+<ComponentUsingRos />
+
+### Battery Charging
+
+When finished with the Jackal, press and release the power button <img src="/img/robot_images/dingo_images/icon-power.png" width="20" /> to power it off.
+
+Jackal's Li-Ion battery pack may be charged internal to the chassis: simply plug in
```
Comment (rhysfaultless-cpr) [inline docs/robots/jackal/user_manual_jackal.mdx]: Suggest changing this to 'lithium battery'

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 9 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Moved all software update info to maintenance as discussed. Made the similar change for Dingo and Husky also.

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 11 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: Fixed

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 10 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: I didn't quite follow what the recommended change was here. Are there switches other than the "NORM" position that need to be set?

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 12 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx:764]: Update to metre for Jackal, Husky, and Dingo.

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 13 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: As discussed, I removed the Windows 10 section entirely.

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 14 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Done

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 15 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Updated for Husky, Jackal, and Dingo

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 16 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Fixed

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 18 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Done

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 19 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Updated for all robots

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 20 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Updated

File: docs/robots/jackal/user_manual_jackal.mdx (hunk 17 above)
Comment (jhiggins-cpr) [reply docs/robots/jackal/user_manual_jackal.mdx]: Done

**Comment by jhiggins-cpr** [comment]
@rhysfaultless-cpr I made most of the updates. Still two open questions (see unresolved threads) where I wasn't sure how to interpret your proposed changes.

File: docs/robots/jackal/tutorials_jackal.mdx (hunk 10 above)
Comment (rhysfaultless-cpr) [reply docs/robots/jackal/tutorials_jackal.mdx]: My mistake. You have told the User to move the switches back to the original positions:
`1. Move the mode switch back to `NORM` (left).`

File: docs/robots/dingo/maintenance_dingo.mdx (hunk 1 above)
Comment (rhysfaultless-cpr) [reply docs/robots/dingo/maintenance_dingo.mdx:7]: I must have missed it in my first review. I'm marking this as resolved.

**Comment by rhysfaultless-cpr** [comment]
Everything looks good. I have marked the two outstanding threads as resolved. Feel free to merge this.

**Pull request merged**